mitmproxy/optmanager.py

```diff
@@ -288,60 +293,77 @@ def __repr__(self):
             options=options
         )
 
-    def set(self, *spec, defer=False):
+    def set(self, *specs: str, defer: bool = False) -> None:
         """
             Takes a list of set specification in standard form (option=value).
             Options that are known are updated immediately. If defer is true,
             options that are not known are deferred, and will be set once they
             are added.
         """
-        vals = {}
-        unknown: typing.Dict[str, typing.List[str]] = collections.defaultdict(list)
-        for i in spec:
-            parts = i.split("=", maxsplit=1)
-            if len(parts) == 1:
-                optname, optval = parts[0], None
+        unprocessed: typing.Dict[str, typing.List[str]] = {}
+        for spec in specs:
+            if "=" in spec:
+                name, value = spec.split("=", maxsplit=1)
+                unprocessed.setdefault(name, []).append(value)
             else:
-                optname, optval = parts[0], parts[1]
-            if optname in self._options:
-                vals[optname] = self.parse_setval(self._options[optname], optval, vals.get(optname))
-            else:
-                unknown[optname].append(optval)
-        if defer:
-            self.deferred.update(unknown)
-        elif unknown:
-            raise exceptions.OptionsError("Unknown options: %s" % ", ".join(unknown.keys()))
-        self.update(**vals)
+                unprocessed.setdefault(spec, [])
 
-    def process_deferred(self):
+        processed: typing.Dict[str, typing.Any] = {}
+        for name in list(unprocessed.keys()):
```

Comment:
maybe it would be easier to understand if this is done in a single loop?

Reply:
We could do that, but for string sequences we'd then need to feed the current value into `parse_setval` as well. I've actually did that first (based on the code that was there), but felt it was much harder to reason about. I've now added feedback that hopefully makes the code a bit easier to understand here.

Reply:
thanks, now it's straightforward 👍